### ISSUE TITLE
New rule: no-line-break-in-string-concatenation

### ISSUE THREAD

**Issue by gajus** [eslint/eslint#6033]
Warns when a newline is added in a string concatenation.

> When the rules will warn. Include a description as well as sample code.

Will warn:

```js
const foo = 'https://www.gandi.net/?' +
        querystring.stringify(query);
```

```js
const foo = 'https://www.gandi.net/?' +
        'foo=FOO&' + 
        'bar=BAR'
```

Will not warn:

```js
const foo = 'https://www.gandi.net/?' + querystring.stringify(query);
```

```js
const foo = 'https://www.gandi.net/?foo=FOO&bar=BAR';
```

> Whether the rule prevents an error or is stylistic.

Stylistic.

> Why the rule should be in the core instead of creating a custom rule.

Modern IDEs allow soft-wrap. Soft-wrap over hard-line enables simpler version control. Hard wrapping need not to be added only for code readability purpose.

> Are you willing to create the rule yourself?

No.

**Comment by alberto** [comment]
Thank for contributing, but I don't think this is general enough for core. Since this hasn't get much traction, I'm closing it.
